src/transaction/log_replication_atomic.cpp

```diff
@@ -248,6 +254,14 @@ namespace cublog
   log_lsa
   atomic_replicator::get_lowest_unapplied_lsa () const
   {
+    std::lock_guard<std::mutex> lockg (m_lowest_unapplied_lsa_mutex);
     return m_lowest_unapplied_lsa;
   }
+
+  void
+  atomic_replicator::set_lowest_unapplied_lsa (log_lsa value)
+  {
+    std::lock_guard<std::mutex> lockg (m_lowest_unapplied_lsa_mutex);
+    m_lowest_unapplied_lsa = value;
```

Comment:
Perform the reconcile calculation between the value returned by the helper and `m_redo_lsa` here. Pseudo:
```
assert (m_redo_lsa is not NULL_LSA);
const LOG_LSA helper_lowest_unapplied_lsa = ....from helper
m_lowest_unapplied_lsa = std::min ( m_redo_lsa, helper_lowest_unapplied_lsa;
```
This code is executed in the - one - replication thread. So, only access to `m_lowest_unapplied_lsa` needs to be synched. The calculation itself can be done before the lock:
```
// .. calculation
{ // .. lock scope
  // lock
  // set
} // .. end lock scope
```